sellable/src/execute.rs

```diff
@@ -58,7 +59,75 @@ where
         Ok(Response::new().add_attribute("method", "list"))
     }
 
-    pub fn try_buy(&mut self, deps: &mut DepsMut, info: MessageInfo) -> Result<Response, ContractError> {
+    pub fn try_buy_token(
+        &mut self,
+        deps: &mut DepsMut,
+        info: MessageInfo,
+        token_id: String,
+    ) -> Result<Response, ContractError> {
+        // check if enough fee was sent
+        if info.funds.len() == 0 {
+            return Err(ContractError::NoFundsPresent);
+        } else if info.funds.len() > 1 {
+            return Err(ContractError::MultipleFundsError);
+        } else {
+            let token = self
+                .listed_tokens
+                .load(deps.storage, token_id.as_str())
+                .map_err(|_| ContractError::NoListedTokensError);
+            match token {
```

Comment:
This match can be replaced with a flat_map e.g.

```rust
self
  .listed_tokens
  .load(deps.storage, token_id.as_str())
  .map_err(|_| ContractError::NoListedTokensError)
  .flat_map(|price| {
    // normal logic here where you can return Err or Ok
  })

Reply:
the return type of the load is a Result<Coin, Error> 
unlike it name "listed_tokens", it only loads a single token